### PR TITLE
Ft admin view users assign roles #187300107

### PR DESCRIPTION
**Pull request by manzidavidrw**
#### What does this PR do?

Admin can view all registered Users 
Admin can assign roles to users 

#### Description of Task to be completed?

- Admin would be able to see the list of all registered Users
- Admin would be able to assign roles to the Users

#### How should this be manually tested?

- you should login first with Admin credentials to be able to perform any of these tasks

#### What are the relevant pivotal tracker/Trello stories?

- id of the project
#187300107

### CODE REVIEW

**Review by Ruthuwamahoro — APPROVED** [review]

File: src/app/(Authentication)/login/page.tsx [hunk 1]
```diff
@@ -98,7 +98,7 @@ export default function Home() {
         showSuccess('Login Successful!');
         setTimeout(() => {
           setPageLoading(true);
-          router.push('/');
+          router.push('/dashboard');
```
Comment (fab-ryan) [inline src/app/(Authentication)/login/page.tsx]: you changed how login should worker

File: src/app/assignrole/page.tsx [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+import { RootState } from '@/redux/store';
+import { useAppDispatch } from '@/redux/hooks/hook';
+import useToast from '@/components/alerts/Alerts';
+import { ToastContainer } from 'react-toastify';
+import { Button, ButtonStyle } from '@/components/formElements';
+import { AssignRoleFields, RoleKeys } from '@/utils/assingRoleFormField';
+import SideNav from '@/components/dashboard/SideNav';
+
+const AssignRoleForm: React.FC = () => {
+  const dispatch = useAppDispatch();
```
Comment (fab-ryan) [inline src/app/assignrole/page.tsx:19]: we have already defined this dispatch which means you will get your function from dispatch

File: src/app/assignrole/page.tsx [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+  const handleSubmit = async (e: FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    const result = await dispatch(
+      assignRole({
+        userId: formState[RoleKeys.UserId],
+        roleId: formState[RoleKeys.RoleId]
+      })
+    );
+    if (assignRole.fulfilled.match(result)) {
```
Comment (fab-ryan) [inline src/app/assignrole/page.tsx]: and again use states from redux not here

File: src/app/dashboard/page.tsx [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+  name: string;
+  email: string;
+  address: string;
+  phone: string;
+}
+
+export default function AdminPanel() {
+  const dispatch = useAppDispatch();
+  const { showSuccess, showError } = useToast();
+
```
Comment (fab-ryan) [inline src/app/dashboard/page.tsx]: and again here

File: src/app/dashboard/page.tsx [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+  if (loading) return <div>Loading...</div>;
+  if (error) return <div>Error: {error}</div>;
+
+  // Pagination logic
+  const indexOfLastUser = currentPage * usersPerPage;
+  const indexOfFirstUser = indexOfLastUser - usersPerPage;
+
+  // Check if users.data is defined before accessing length
+  const totalUsers = users?.data ? users.data.length : 0;
+
```
Comment (fab-ryan) [inline src/app/dashboard/page.tsx:43]: remember to remove console.log

File: src/redux/slices/assignroleSlice.ts [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+        state.loading = false;
+        state.success = true;
+        state.error = null;
+      })
+      .addCase(assignRole.rejected, (state, action) => {
+        state.loading = false;
+        state.success = false;
+        state.error = action.payload
+          ? (action.payload as any).message
+          : 'Unknown error';
```
Comment (fab-ryan) [inline src/redux/slices/assignroleSlice.ts:79]: why can't you use this states

**Review by fab-ryan — CHANGES_REQUESTED** [review]
test every each corner of your task

**Review by fab-ryan — APPROVED** [review]

**Review by RWEMAREMY — APPROVED** [review]
great work

**Review by Mukunzijames — APPROVED** [review]

**Pull request merged**